spec/plurimath/math_zone_spec.rb

```diff
@@ -0,0 +1,296 @@
+                  |_ "theta" base expression
+        MATHZONE
+        match_puts(expected_value)
+      end
+    end
+
+    context "AsciiMath Math zone representation of color #11" do
+      let(:exp) { 'color(red)(theta)' }
+
+      it 'should puts Math zone representation of sample example #11' do
+        expected_value = <<~MATHZONE
+          |_ Math zone
+            |_ "color(red)(theta)"
+               |_ "color(red)(theta)" color
+                  |_ "r e d" mathcolor
+                  |_ "theta" text
+        MATHZONE
+        match_puts(expected_value)
+      end
+    end
+
+    context "AsciiMath Math zone representation of four unary functions #12" do
+      let(:exp) { 'abs(x)floor(x)ceil(x)norm(vecx)' }
+
+      it 'should puts Math zone representation of sample example #12' do
+        expected_value = <<~MATHZONE
+          |_ Math zone
+            |_ "abs(x) floor(x) ceil(x) norm(vec(x))"
+               |_ "abs(x)" function apply
+               |  |_ "abs" function name
+               |  |_ "x" argument
+               |_ "floor(x)" function apply
+               |  |_ "floor" function name
+               |  |_ "x" argument
+               |_ "ceil(x)" function apply
+               |  |_ "ceil" function name
+               |  |_ "x" argument
+               |_ "norm(vec(x))" function apply
+                  |_ "norm" function name
+                  |_ "vec(x)" argument
+                     |_ "vec(x)" function apply
+                        |_ "vec" function name
+                        |_ "x" argument
+        MATHZONE
+        match_puts(expected_value)
+      end
+    end
+
+    context "AsciiMath Math zone representation of font functions #13" do
+      let(:exp) { 'mathbf(x)mathbb(x)mathcal(x)mathtt(x)mathfrak(x)mathsf(x)' }
+
+      it 'should puts Math zone representation of sample example #13' do
+        expected_value = <<~MATHZONE
+          |_ Math zone
+            |_ "mathbf(x) mathbb(x) mathcal(x) mathtt(x) mathfrak(x) mathsf(x)"
+               |_ "mathbf(x)" function apply
+               |  |_ "mathbf" function name
+               |  |_ "x" argument
+               |_ "mathbb(x)" function apply
+               |  |_ "mathbb" function name
+               |  |_ "x" argument
+               |_ "mathcal(x)" function apply
+               |  |_ "mathcal" function name
+               |  |_ "x" argument
+               |_ "mathtt(x)" function apply
+               |  |_ "mathtt" function name
+               |  |_ "x" argument
+               |_ "mathfrak(x)" function apply
+               |  |_ "mathfrak" function name
+               |  |_ "x" argument
+               |_ "mathsf(x)" function apply
+                  |_ "mathsf" function name
+                  |_ "x" argument
+        MATHZONE
+        match_puts(expected_value)
+      end
+    end
+  end
+end
+
+def match_puts(expected_value)
+  expect { formula }.to output(expected_value).to_stdout
```

Comment:
Need to change `to_stdout` in accordance with the return of a string from the new `#to_display` method.